packages/grid/_modules_/grid/models/api/densityApi.ts

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { DensityTypes, Density } from '../gridOptions';
+
+export interface DensityConfig {
+  density: Density;
+  headerHeight: number;
+  rowHeight: number;
+}
+
+export interface DensityOption {
+  icon: React.ReactElement;
+  label: DensityTypes;
+}
+
+/**
+ * The density API interface that is available in the grid [[apiRef]].
+ */
+export interface DensityApi {
+  /**
+   * Set density of the grid.
+   * @param density
+   * @returns void
+   */
+  setDensity: (density: Density, headerHeight?, rowHeight?) => void;
```

Comment:
ShowDensityMenu, HideDensityMenu

Reply:
Do we really need API methods for those, It's just a toggle of a boolean. My argument is if people want to have their own custom density picker they will also most likely have their own menu and the functionality to open and close it.

IMO the part of the density picker that makes sense to be a public API is the way to change it since this is also the part of the functionality that affects the state and the grid in general.

Reply:
I'm personally in favor of removing all imperative APIs, and to only add them, on a case by case basis if asked by developers with a solid case.